ignite/contrib/handlers/mlflow_logger.py

```diff
@@ -231,30 +229,26 @@ def __call__(self, engine: Engine, logger: MLflowLogger, event_name: Union[str,
                 " Please check the output of global_step_transform."
             )
 
-        rendered_metrics = {}  # type: Dict[str, float]
-        for key, value in metrics.items():
-            if isinstance(value, numbers.Number):
-                rendered_metrics[f"{self.tag} {key}"] = value  # type: ignore[assignment]
-            elif isinstance(value, torch.Tensor) and value.ndimension() == 0:
-                rendered_metrics[f"{self.tag} {key}"] = value.item()
-            elif isinstance(value, torch.Tensor) and value.ndimension() == 1:
-                for i, v in enumerate(value):
-                    rendered_metrics[f"{self.tag} {key} {i}"] = v.item()
-            else:
-                warnings.warn(f"MLflowLogger output_handler can not log metrics value type {type(value)}")
-
         # Additionally recheck metric names as MLflow rejects non-valid names with MLflowException
         from mlflow.utils.validation import _VALID_PARAM_AND_METRIC_NAMES
 
-        for key in list(rendered_metrics.keys()):
+        for keys in list(rendered_metrics.keys()):
+            key = " ".join(keys)
+            print(key, keys)
             if not _VALID_PARAM_AND_METRIC_NAMES.match(key):
                 warnings.warn(
                     f"MLflowLogger output_handler encountered an invalid metric name '{key}' that "
                     "will be ignored and not logged to MLflow"
                 )
-                del rendered_metrics[key]
+                del rendered_metrics[keys]
+
+        metrics = {}
+        for key, value in rendered_metrics.items():
+            print(key)
+            key = " ".join(key)
+            metrics[key] = value
```

Comment:
Let's do this step before the previous one where we can remove invalid keys. Such that we call `key = " ".join(key)` once

Reply:
Yeah makes sense, fixed!